### PR TITLE
Init

### PR DESCRIPTION
**Pull request by iruzevic**
# Description

- extracting funkcionality.

### CODE REVIEW

**Review by mbmjertan — DISMISSED** [review]
looks okay

**Review by goranalkovic-infinum — DISMISSED** [review]

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Project Specific
+/vendor
+/.vscode
```
Comment (piqusy) [inline .gitignore:3]:
```suggestion
/.vscode

# Mac OS custom attribute store and thumbnails
*.DS_Store
._*
```

File: README.md [hunk 2]
```diff
@@ -1 +1,42 @@
...
+![image](https://repository-images.githubusercontent.com/204961525/1788eeba-8046-47bf-b315-32b060db1c8e)
+
+[![GitHub tag](https://img.shields.io/github/tag/infinum/eightshift-forms-utils.svg?style=for-the-badge)](https://github.com/infinum/eightshift-forms-utils)
+[![GitHub stars](https://img.shields.io/github/stars/infinum/eightshift-forms-utils.svg?style=for-the-badge&label=Stars)](https://github.com/infinum/eightshift-forms-utils)
+[![license](https://img.shields.io/github/license/infinum/eightshift-forms-utils.svg?style=for-the-badge)](https://github.com/infinum/eightshift-forms-utils)
+[![codecov](https://img.shields.io/codecov/c/gh/infinum/eightshift-forms-utils/branch/develop?label=Codecov&style=for-the-badge&token=02Lfa2jjoK)](https://codecov.io/gh/infinum/eightshift-forms-utils)
+
+# Eightshift Forms Utils Library
+
+Eightshift Forms Utils library is a set of helpers and utils that can be used in the main Eightshift Forms plugin or additional Add-on plugins.
```
Comment (piqusy) [inline README.md:10]:
```suggestion
Eightshift Forms Utils library is a set of helpers and utils that can be used in the main Eightshift Forms plugin or additional Add-ons.
```

File: src/Blocks/UtilsBlocks.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+declare(strict_types=1);
+
+namespace EightshiftFormsUtils\Blocks;
+
+use EightshiftFormsUtils\Helpers\UtilsHooksHelper;
+use EightshiftLibs\Blocks\AbstractBlocks;
+
+/**
+ * Class AbstractUtilsBaseRoute
```
Comment (piqusy) [inline src/Blocks/UtilsBlocks.php:18]:
```suggestion
 * Class UtilsBlocks
```

File: src/Blocks/UtilsBlocks.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
+<?php
+
+/**
+ * Class Blocks is the base class for Gutenberg blocks registration.
```
Comment (piqusy) [inline src/Blocks/UtilsBlocks.php:4]:
```suggestion
 * Class UtilsBlocks is the base class for Gutenberg blocks registration.
```

File: src/Config/UtilsConfig.php [hunk 5]
```diff
@@ -0,0 +1,563 @@
...
+	 * @var string
+	 */
+	public const ROUTE_VERSION = 'v1';
+
+	public const API_RESPONSE_CODE_SUCCESS = 200;
+	public const API_RESPONSE_CODE_SUCCESS_RANGE = 299;
+	public const API_RESPONSE_CODE_ERROR = 400;
+	public const API_RESPONSE_CODE_ERROR_MISSING = 404;
+	public const API_RESPONSE_CODE_ERROR_SERVER = 500;
+	public const API_RESPONSE_CODE_ERROR_FORBIDDEN = 403;
```
Comment (piqusy) [inline src/Config/UtilsConfig.php:448]:
```suggestion
	public const API_RESPONSE_CODE_ERROR = 400;
	public const API_RESPONSE_CODE_ERROR_FORBIDDEN = 403;
	public const API_RESPONSE_CODE_ERROR_MISSING = 404;
	public const API_RESPONSE_CODE_ERROR_SERVER = 500;
```

**Review by goranalkovic-infinum — APPROVED** [review]

File: src/Helpers/UtilsGeneralHelper.php [hunk 6]
```diff
@@ -0,0 +1,923 @@
...
+			}
+
+			$output[$name] = $value;
+		}
+
+		return $output;
+	}
+
+	/**
+	 * Return counries filtered by some key for multiple usages.
```
Comment (piqusy) [inline src/Helpers/UtilsGeneralHelper.php:669]:
```suggestion
	 * Return countries filtered by some key for multiple usages.
```

File: src/Helpers/UtilsGeneralHelper.php [hunk 7]
```diff
@@ -0,0 +1,923 @@
...
+			if ($key === $needle) {
+				\array_push($aHitList, $value);
+			}
+		}
+
+		return $aHitList;
+	}
+
+	/**
+	 * Output select options ass array from html string.
```
Comment (piqusy) [inline src/Helpers/UtilsGeneralHelper.php:730]:
```suggestion
	 * Output select options as array from HTML string.
```

File: src/Helpers/UtilsGeneralHelper.php [hunk 8]
```diff
@@ -0,0 +1,923 @@
...
+	 */
+	public static function canIntegrationUseSync(string $integrationName): bool
+	{
+		return isset(\apply_filters(UtilsConfig::FILTER_SETTINGS_DATA, [])[$integrationName]['fields']);
+	}
+
+	/**
+	 * Clean url from query params.
+	 *
+	 * @param string $url Url to clean.
```
Comment (piqusy) [inline src/Helpers/UtilsGeneralHelper.php:843]:
```suggestion
	 * Clean URL from query params.
	 *
	 * @param string $url URL to clean.
```

File: src/Helpers/UtilsSettingsOutputHelper.php [hunk 9]
```diff
@@ -0,0 +1,334 @@
...
+	 */
+	public static function getMiscDisclaimer(): array
+	{
+		return [
+			'component' => 'layout',
+			'layoutType' => 'layout-v-stack-card',
+			'layoutContent' => [
+				[
+					'component' => 'intro',
+					'introTitle' => \__('Disclamer', 'eightshift-forms'),
```
Comment (piqusy) [inline src/Helpers/UtilsSettingsOutputHelper.php:117]:
```suggestion
					'introTitle' => \__('Disclaimer', 'eightshift-forms'),
```

**Review by piqusy — COMMENTED** [review]
Haven't reviewed everything bc the PR got merged while reviewing 😄

**Pull request merged**